js/models/question-component-statement.js

```diff
@@ -0,0 +1,90 @@
+define(function(require) {
+
+  require('../xapiwrapper.min');
+
+  var Adapt = require('coreJS/adapt');
+  var _ = require('underscore');
+  var Backbone = require('backbone');
+  var ComponentStatementModel = require('./component-statement');
+
+  return ComponentStatementModel.extend({
+
+    initialize: function() {
+      return ComponentStatementModel.prototype.initialize.call(this);
+    },
+
+    getStatement: function() {
+      var statement = ComponentStatementModel.prototype.getStatement.call(this);
+
+      var verb = this.getVerb();
+      var object = this.getObject();
+      var context = this.getContext();
+      var result = this.getResult();
+
+      if (
```

Comment:
This `if` statement appears all over the place.  It should probably be abstracted into a common function.